### DISCUSSION TITLE
Crazyflie Flowdeck Messing Up

### DISCUSSION THREAD

**Discussion by akshaysunkara** [bitcraze/discussions#1480] [Q&A]
Hey guys, whenever I'm trying to fly my drone, it really messes up and tilts on one side of the drone.

I noticed a problem when the drone is still, the values of X and Y just keep going up even though its not moving at all:
https://github.com/user-attachments/assets/482b25d6-cdb4-481d-ad9c-61932dbfb800

This is how the drone flies:
https://github.com/user-attachments/assets/9064fe73-93d7-429d-a7a9-0be6fb7b9217

By the way, the x and y coordinates in the state estimate go up even if the flowdeck isn't attached. Is there like a way to recalibrate whichever component is responsible for the coordinates? 

Thanks again

**Comment by knmcguire** [comment]
Hi! So this happens after you have fixed the flowdeck in https://github.com/orgs/bitcraze/discussions/1473 right?

Could you share the console output? The deck is probably been detected but there is definitely still something wrong.

If you plot the motion.x/y logging variable in the plotting tab of the cfclient, and just hold it in your hand and move it above the floor, do you see anything happening?

**Reply by akshaysunkara** [reply]
Hey, after checking this morning, the X Y thing seems fine and stays at 0 when I don't attach the flowdeck.

After attaching the flowdeck, the crazyflie just starts blinking red three times, pauses, and does it again for M1 instead of blinking red once every second. I'm not able to connect to the crazyflie now that the flowdeck is attached.

And yes, this was after I fixed the flowdeck in that discussion

**Reply by akshaysunkara** [reply]
Hey, I just figured out how to attach the flowdeck. 

I'm able to get it to connect but it still isn't able to fly.

This is a recording of the cfclient gui whenever I try flying it. Even when I don't fly it, the Z value keeps going up.

https://github.com/user-attachments/assets/007f3694-41f4-4c0a-969e-f16306c86e03

The drone starts to tip over to its left side.

These are the console logs of two trials whenever I try attaching the flowdeck now. For some reason, its not recognizing it anymore. There's just so many issues with it:

ESTKALMAN: State out of bounds, resetting
SYS: Crazyflie 2.1 is up and running!
SYS: Build 0:e527895e8c6e (2023.11) CLEAN
SYS: I am 0x31323231323247130033003E and I have 1024KB of flash!
CFGBLK: v1, verification [OK]
DECK_INFO: Memory error: incorrect TLV CRC 69!=FF
DECK_INFO: Deck 0 has corrupt OW memory. No driver will be initialized!
DECK_CORE: 0 deck(s) found
IMU: BMI088: Using I2C interface.
IMU: BMI088 Gyro connection [OK].
IMU: BMI088 Accel connection [OK]
IMU: BMP388 I2C connection [OK]
ESTIMATOR: Using Complementary (1) estimator
CONTROLLER: Using PID (1) controller
MTR-DRV: Using brushed motor driver
SYS: About to run tests in system.c.
SYS: NRF51 version: 2024.2 (CF21)
EEPROM: I2C connection [OK].
STORAGE: Storage check [OK].
IMU: BMI088 gyro self-test [OK]
SYS: Self test passed!
STAB: Wait for sensor calibration...
SYS: Free heap: 20496 bytes
STAB: Starting stabilizer loop
SUP: Can not fly
SUP: Ready to fly
SUP: Can not fly
SUP: Ready to fly
SUP: Can not fly
SUP: Ready to fly
SUP: Can not fly
SUP: Ready to fly
SUP: Can not fly
SUP: Ready to fly
SUP: Can not fly
SUP: Ready to fly
SUP: Can not fly
SUP: Ready to fly
SUP: Can not fly
SUP: Ready to fly
SUP: Can not fly
SUP: Ready to fly
SYS: ----------------------------
SYS: Crazyflie 2.1 is up and running!
SYS: Build 0:e527895e8c6e (2023.11) CLEAN
SYS: I am 0x31323231323247130033003E and I have 1024KB of flash!
CFGBLK: v1, verification [OK]
DECK_CORE: 0 deck(s) found
IMU: BMI088: Using I2C interface.
IMU: BMI088 Gyro connection [OK].
IMU: BMI088 Accel connection [OK]
IMU: BMP388 I2C connection [OK]
ESTIMATOR: Using Complementary (1) estimator
CONTROLLER: Using PID (1) controller
MTR-DRV: Using brushed motor driver
SYS: About to run tests in system.c.
SYS: NRF51 version: 2024.2 (CF21)
EEPROM: I2C connection [OK].
STORAGE: Storage check [OK].
IMU: BMI088 gyro self-test [OK]
SYS: Self test passed!
STAB: Wait for sensor calibration...
SYS: Free heap: 20496 bytes

I'm going to try soldering the flowdeck or something today to get it to attach but would that solve the issue that I'm having? Is there another component that's fried or something?

**Reply by akshaysunkara** [reply]
Yeah, I can confirm that it is for sure a flowdeck problem - I just checked with another drone that was working with another flowdeck but it wasn't working properly. 

These are the pictures of the flowdeck:
![Untitled](https://github.com/user-attachments/assets/f7cc18be-b8b4-40f7-951e-11b6c17c3070)

![Untitled-1](https://github.com/user-attachments/assets/d3171c5b-e911-4423-bff8-4272b23913a2)

The pins to attach them are gone but I still make the pins go through the hole.

Does it seem like there's anything wrong with the flowdeck?

**Reply by knmcguire** [reply]
That is a big problem. You need the female connectors in order for the pins to make right contact consistently all the time. Now the pins will continously connect and disconnect during flight so I'm pretty sure that is causing your issues.

You can consider soldering the pins on the flowdeck directly like in here: https://www.bitcraze.io/documentation/tutorials/solder-deck-pins/

Or to order new female pin headers from the shop and repair those: https://store.bitcraze.io/collections/spare-parts/products/female-deck-connector

Also in the future, make sure to always mount the decks straight on the pins and never remove it by rotating it to one side. This will cause the female deck connectors to get damaged or get ripped of like that.
